h2quic/server.go

```diff
@@ -88,6 +89,10 @@ func (s *Server) serveImpl(tlsConfig *tls.Config, conn net.PacketConn) error {
 		return errors.New("use of h2quic.Server without http.Server")
 	}
 	s.listenerMutex.Lock()
+	if s.closed {
```

Comment:
Can we defer unlock here?

Reply:
No, because the loop at the end of this function will block until the server is closed.

Reply:
Sorry, GH didn't show the whole file 🙄